SpatialGDK/Source/SpatialGDKFunctionalTests/Public/TestMaps/GeneratedTestMap.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "GenerateTestMapsCommandlet.h"
+#include "GeneratedTestMap.generated.h"
+
+/**
+ * The base class for automatically generating test maps.
+ * Descend from this and your class should automatically be picked up by the GenerateTestMapsCommandlet for generation.
+ */
+UCLASS()
+class SPATIALGDKFUNCTIONALTESTS_API UGeneratedTestMap : public UObject
+{
+	GENERATED_BODY()
+
+public:
+	UGeneratedTestMap();
+
+	virtual void GenerateMap();
+	virtual bool ShouldGenerateMap() { return true; } // To control whether to generate a map from this class
+	virtual FString GetMapName() { return MapName; }
+	static FString GetGeneratedMapFolder();
+
+protected:
+	virtual void GenerateBaseMap();
+	virtual void SaveMap();
+	// This is what test maps descending from this class should normally override to add their own content to the map
+	virtual void CreateCustomContentForMap() {}
+
+	virtual FString GetPathToSaveTheMap();
+
+	enum EMapCategory
```

Comment:
These values could use some comments about what they mean. Unless we think that doubling up the docs between code comments and our wiki will mean one or the other will get outdated - in this case we could at least link to the info in the wiki here?